OgmentoAPI.Domain.Catalog.Infrastructure/Repository/CategoryRepository.cs

```diff
@@ -0,0 +1,215 @@
+			return category.ID;
+		}
+		public List<Category> GetAllParentCategories()
+		{
+			return _dbContext.Category.AsNoTracking().Where(x => x.ParentCategoryId == 1).ToList();
+		}
+		public async Task<Guid> GetParentGuidFromParentId(int? parentId) { 
+			if(parentId == null)
+			{
+				return _dbContext.Category.AsNoTracking().FirstOrDefault(x => x.CategoryName == "All Products").CategoryUid;
+			}
+			return _dbContext.Category.AsNoTracking().FirstOrDefault(x => x.ID == parentId).CategoryUid;
+		}
+		public List<CategoryModel> GetSubCategoriesByCategoryId(int Id)
+		{
+
+			List<Category> subCategories = _dbContext.Category.AsNoTracking().Where(x => x.ParentCategoryId == Id).ToList();
+			List<CategoryModel> categoryModel = subCategories.Select(x => new CategoryModel
+			{
+				CategoryName = x.CategoryName,
+				ID = x.ID,
+				CategoryUid = x.CategoryUid,
+				ParentCategoryId = x.ParentCategoryId,
+				ParentCategoryUid = GetParentGuidFromParentId(x.ParentCategoryId).Result,
+				SubCategories = SubCategoriesCount(x.ID) != 0 ? GetSubCategoriesByCategoryId(x.ID) : null,
+
+			}).ToList();
+			return categoryModel;
+
+		}
+		
+		private int SubCategoriesCount(int Id)
+		{
+			return _dbContext.Category.Count(x => x.ParentCategoryId == Id);
+		}
+
+		public List<CategoryModel> GetCategoriesByProductId(int productId)
+		{
+			//I will change this. I know this has issues.
+			List<int> categoryIds = _dbContext.ProductCategoryMapping.Where(x => x.ProductId == productId).Select(x => x.CategoryId).ToList();
+			List<Category> categories = _dbContext.Category.Where(x => categoryIds.Contains(x.ID)).ToList();
+			List<CategoryModel> categoryModels = categories.Select(x => new CategoryModel
+			{
+				ID = x.ID,
+				CategoryName = x.CategoryName,
+				CategoryUid = x.CategoryUid,
+				ParentCategoryId = x.ParentCategoryId,
+				ParentCategoryUid = GetParentGuidFromParentId(x.ParentCategoryId).Result,
+				SubCategories = SubCategoriesCount(x.ID) != 0 ? GetSubCategoriesByCategoryId(x.ID) : null,
+			}).ToList();
+			return categoryModels;
+
+		}
+		public CategoryModel GetCategoryByCategoryId(int? categoryId)
+		{
+			Category category = _dbContext.Category.FirstOrDefault(x => x.ID == categoryId);
+			CategoryModel categoryModel = new CategoryModel()
+			{
+				ID = category.ID,
+				CategoryName = category.CategoryName,
+				CategoryUid = category.CategoryUid,
+				ParentCategoryId = category.ParentCategoryId,
+				ParentCategoryUid = GetParentGuidFromParentId(category.ParentCategoryId).Result,
+				SubCategories = GetSubCategoriesByCategoryId(category.ID),
+			};
+			return categoryModel;
+		}
+		private bool IsSafeDelete(int categoryId)
+		{
+			if (_dbContext.ProductCategoryMapping.Any(x => x.CategoryId == categoryId))
+			{
+				return false;
+			}
+			return true;
+		}
+		public async Task DeleteCategory(int? categoryId)
+		{
+			Category category = _dbContext.Category.FirstOrDefaultAsync(x => x.ID == categoryId).Result;
```

Comment:
```suggestion
			Category category = _dbContext.Category.FirstOrDefault(x => x.ID == categoryId);
```
When not using async await, directly use th eother method